src/System.Net.Http/src/System/Net/Http/HttpRequestException.cs

```diff
@@ -31,10 +31,17 @@ public HttpRequestException(string message, Exception inner)
 
         // This constructor is used internally to indicate that a request was not successfully sent due to an IOException,
         // and the exception occurred early enough so that the request may be retried on another connection.
-        internal HttpRequestException(string message, Exception inner, bool allowRetry)
+        internal HttpRequestException(string message, Exception inner, RequestRetryType allowRetry)
             : this(message, inner)
         {
             AllowRetry = allowRetry;
         }
     }
+
+    internal enum RequestRetryType
+    {
+        NoRetry,
+        RetryOnSameProxy,
+        RetryOnNextProxy
+    }
```

Comment:
I think it'd be better to either a) put this in its own file, or b) make it a nested type of HttpRequestException.